internal/app/services/core/payments/payment_usecase_impl.go

```
@@ -1222,158 +1496,64 @@ func (e *resourceFetchError) Unwrap() error { return e.err }
+			err,
+			constvars.StatusBadRequest,
+			"Patient ID does not match with the current user",
+			"precondition: patient mismatch",
 		)
 	}
+	return fetchedPatient, nil
+}
 
-	if fetchedSlot.Status != fhir_dto.SlotStatusFree {
+// validateSlotEligibility checks slot status, schedule ownership, and that the slot is in the future.
+func (_ *paymentUsecase) validateSlotEligibility(slot *fhir_dto.Slot, schedules []fhir_dto.Schedule, schedulesErr error, slotID, _ string) (*fhir_dto.Schedule, error) {
```

golangci-lint (2.12.2)

[error] ST1006: receiver name should not be an underscore, omit the name if it is unused

(staticcheck)